docs/start/install.mdx

```diff
@@ -137,6 +137,121 @@ Additionally, Qiskit only supports CPython. Running with other Python interprete
+as not being considered stable interfaces yet and a user visible warning will be
+actively emitted on any use of these unstable interfaces. Additionally, in some
+situations an interface marked as private will be considered part of the public
+API. Typically this only occurs in two cases, either an abstract interface
+definition where subclasses are intended to override/implement a private method
+as part of defining an implementation of the interface, or advanced-usage
+low-level methods that have stable interfaces but are not considered safe to use
+as the burden is on the user to uphold the class/safety invariants themselves
+(the canonical example of this is the `QuantumCircuit._append` method)
+
+The supported Python versions, minimum supported Rust version (for building
+Qiskit from source), and any Python package dependencies (including the minimum
+supported versions of dependencies) used by Qiskit are not part of the backwards
+compatibility guarantees and may change during any release. Only minor or major
+version releases will raise minimum requirements for using or building Qiskit
+(including adding new dependencies), but patch fixes might include support for
+new versions of Python or other dependencies. Usually the minimum version of a
+dependency is only increased when older dependency versions go out of support or
+when it is not possible to maintain compatibility with the latest release of the
+dependency and the older version.
+
+### Upgrade Strategy
+
+Whenever a new major version is released, the recommended upgrade path
+is to first upgrade to use the most recent minor version on the previous major
+version. Immediately preceding a new major version a final minor version will
+be published. This final minor version release `M.N+1.0` is equivalent to
+`M.N.0` but with warnings and deprecations for any API changes that are
+made on the new major version series.
+
+For example, immediately proceeding the 1.0.0 release a 0.46.0 release will be
+published. The 0.46.0 release will be equivalent to the 0.45.0 release but with
+additional deprecation warnings that document the API changes that were made as
+part of the 1.0.0 release. This pattern will be used for any future major
+version releases.
+
+As a user of Qiskit, it's recommended that you first upgrade to this final minor
+version first, so you can see any deprecation warnings and adjust your Qiskit
+usage ahead of time before trying a potentially breaking release. The previous
+major version will be supported for at least 6 months to give sufficient time
+to upgrade. A typical pattern to deal with this is to pin the max version to
+avoid using the next major release series until you're sure of compatibility.
+For example, specifying in a requirements file `qiskit<2` when the current
+major Qiskit version is 1 will ensure that you're using a version of Qiskit
+that won't have breaking API changes.
```

Comment:
I think a "Qiskit as a dependency" can be its own section, to give more relevance and detail (for example, downstream using a feature introduced in a minor should use `~=`.)